### PR TITLE
DynatraceResourceDetector: Change log level to debug

### PR DESCRIPTION
**Pull request by joaopgrassi**
Commit Message: Stop emitting warning logs when the Dynatrace resource detector doesn't file a file to load attributes from.

Additional Description: It is expected that some files are not present, so the warning log is too verbose. We already log properly as a warning when NO files are found, which is a erroneous situation.

Risk Level: Low
Testing: Manual
Docs Changes: No
Release Notes: No
Platform Specific Features: No
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by joaopgrassi** [comment]
@wbpcode not a big deal if not, but any chance we can get this backported to the 1.30 release?

File: changelogs/current.yaml [hunk 1]
```diff
@@ -5,6 +5,9 @@ behavior_changes:
 
 minor_behavior_changes:
 # *Changes that may cause incompatibilities for some users, but should not for most*
+- area: tracing
+  change: |
+    Dynatrace resource detector no longer logs a warning when failing to read individual enrichment files.
```
Comment (wbpcode) [inline changelogs/current.yaml:10]: Actually, I think this change log is unnecessary.

**Comment by wbpcode** [comment]
> @wbpcode not a big deal if not, but any chance we can get this backported to the 1.30 release?

Yeah. If you like it.

File: changelogs/current.yaml (hunk 1 above)
Comment (joaopgrassi) [reply changelogs/current.yaml:10]: Yeah I agree. But then CI fails right? Can it be ignored?

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**